### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by amshinde**

### CODE REVIEW

File: src/util.c [hunk 1]
```diff
@@ -573,7 +573,7 @@ cc_oci_enable_networking (void)
  * \return \c guint32 in big endian order.
  */
 guint32 cc_oci_get_big_endian_32(const guint8 *buf) {
-	return (guint32)(buf[0] >> 24 | buf[1] >> 16 | buf[2] >> 8 | buf[3]);
+	return (guint32)(buf[0] << 24 | buf[1] << 16 | buf[2] << 8 | buf[3]);
```
Comment (jodh-intel) [inline src/util.c]: This demonstrates that we really need some unit tests for these functions. Could you look into that on a separate PR?

**Comment by jodh-intel** [comment]
lgtm.

**Comment by dlespiau** [comment]
lgtm

**Comment by chavafg** [comment]
qa-failed

**Comment by amshinde** [comment]
Added a commit to fix file descriptor leaks.

**Comment by sameo** [comment]
LGTM

**Comment by dlespiau** [comment]
lgtm

**Pull request merged**